src/edu/solvd/mentoring/Support.java

```diff
@@ -0,0 +1,417 @@
+  private static short getShort(short maxValue, short minValue){
+    return (short)(Math.round(Math.random() * (maxValue - minValue)) + minValue);
+  }
+  private static HashMap<Integer,Short> getGeneratedClutch(short maxClutch, short minClutch, HashMap<Integer,Surface> surfaces){
+    HashMap<Integer,Short> clutch = new HashMap<>();
+    Set<Integer> surfaceKeys = surfaces.keySet();
+
+    for(Integer i: surfaceKeys){
+      short val = s(getShort(maxClutch,minClutch));
+      clutch.put(i,val);
+    }
+
+    return clutch;
+  }
+
+  private static void initSurfaces(){
+      surfaces.put(1,new Surface("highway",'=',1));
+      surfaces.put(2,new Surface("city road",'-',2));
+      surfaces.put(3,new Surface("ferry road",'+',3));
+      surfaces.put(4,new Surface("country road",'*',4));
+  }
+
+  public static Surface getSurface(int id){
+    if (surfaces.isEmpty()){
+      initSurfaces();
+    }
+    return surfaces.get(id);
+  }
+
+  public static HashMap<Integer, Surface> getSurfaces(){
+    if (surfaces.isEmpty()){
+      initSurfaces();
+    }
+    return surfaces;
+  }
+
+  private static void initWheels(){
+      HashMap<Integer,Short> wheelClutch1 = new HashMap<>();
+      HashMap<Integer,Short> wheelClutch2 = new HashMap<>();
+      HashMap<Integer,Short> wheelClutch3 = new HashMap<>();
+      HashMap<Integer,Short> wheelClutch4 = new HashMap<>();
+      HashMap<Integer,Short> wheelClutch5 = new HashMap<>();
+
+      //Surface id, clutch [0-7]
+      wheelClutch1.put(1,s(2));
+      wheelClutch1.put(2,s(2));
+      wheelClutch1.put(3,s(2));
+      wheelClutch1.put(4,s(2));
+
+      wheelClutch2.put(1,s(5));
+      wheelClutch2.put(2,s(1));
+      wheelClutch2.put(3,s(2));
+      wheelClutch2.put(4,s(1));
+
+      wheelClutch3.put(1,s(1));
+      wheelClutch3.put(2,s(1));
+      wheelClutch3.put(3,s(3));
+      wheelClutch3.put(4,s(4));
+
+      wheelClutch4.put(1,s(1));
+      wheelClutch4.put(2,s(1));
+      wheelClutch4.put(3,s(7));
+      wheelClutch4.put(4,s(7));
+
+      wheelClutch5.put(1,s(4));
+      wheelClutch5.put(2,s(6));
+      wheelClutch5.put(3,s(5));
+      wheelClutch5.put(4,s(3));
+
+
+
+      wheels.put(1,new Wheel(DetailQuality.C,"Basic",wheelClutch1));
+      wheels.put(2,new Wheel(DetailQuality.B,"Sport",wheelClutch2));
+      wheels.put(3,new Wheel(DetailQuality.B,"OffRoad",wheelClutch3));
+      wheels.put(4,new Wheel(DetailQuality.A,"OffRoad II",wheelClutch4));
+      wheels.put(5,new Wheel(DetailQuality.S,"Universal LU",wheelClutch5));
+  }
+
+  public static Wheel getWheel(int id){
+    if(wheels.isEmpty() == true){
+      initWheels();
+    }
+    return wheels.get(id);
+  }
+
+  public static Wheel getGeneratedWheel(HashMap<Integer,Surface> surfaces){
+    HashMap<Integer,Short> clutch = getGeneratedClutch(s(7),s(0),surfaces);
+    int avg = 0;
+    for (short val: clutch.values()){
+      avg += val;
+    }
+    avg = Math.round(avg/clutch.size());
+    DetailQuality dq;
+    switch (avg) {
+      case 1:
```

Comment:
why is it empty?

Reply:
When Support.getSmth(id) calles first time nothing is initialized, soin order not to initialized unnecessary parts, like wheels when we try to get CarBase and not spend time

Reply:
Not sure about java, but
 case 3:
 case 2: 
    a = 2;
    break;
Means that in case 2 and 3 a Will take 2, code goes until the first break

Reply:
actually, usage of switch-case is rare, so it should be tested in java, not sure about this. 
I suppose to make DetailQuality enum which can pass GOOD(1), MEDIUM(0.8), BAD(0.7) and so on and just multiply that coefficient to your detail chars. So you will avoid switch and could make any number of qualities you like without additional code changes

Reply:
I actually thougth in reverse way, that detail quality depends on total characteristicks it has